lib/livebook/evaluator/io_proxy.ex

```diff
@@ -150,6 +156,16 @@ defmodule Livebook.Evaluator.IOProxy do
     io_requests(reqs, {:ok, state})
   end
 
+  # Livebook custom request type, handled in a special manner
+  # by IOProxy and safely failing for any other IO device
+  # (resulting in the {:error, :request} response).
+  defp io_request({:livebook_put_term, term}, state) do
```

Comment:
[ref](https://github.com/jonatanklosko/live_widget/blob/60f9cacd9a7bbb7f0add9773f570056b8d554a6f/lib/live_widget.ex#L51)

Reply:
Honestly, I am not a very big fan of extending the io contract like that. It means things will fail really poorly if it goes to an actual IO process. Maybe we could store something in the process dictionary?

Reply:
Actually a regular IO process would return `{:error, :request}`, which means it won't cause exception in scripts (although we could easily throw an exception from the render function ourselves). I thought this would be in line with the other non-supported operations like `:get_geometry`, except we get a slightly different error because it's not part of the spec.

Reply:
Ah, you are right, there are already extensions to the I/O protocol... so I guess we will be fine.